docs/deployment-gcp.md

```diff
@@ -0,0 +1,134 @@
+# Deploying Canopy on Google Cloud Platform (GCP) - Step-by-Step Guide
+
+
+## Introduction
+In this guide, we'll walk through the process of deploying Canopy on Google Cloud Platform (`GCP`). 
+The steps include setting up GCP, creating a Docker repository, pulling and tagging the `Canopy` image, and finally deploying it using Google Cloud Run.
+
+## Prerequisites
+Before you begin, make sure you have the following installed:
+
+- [Docker](https://docs.docker.com/engine/install/)
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install)
+
+and make sure you have a project set up in `GCP`.
+
+## Step 1: Install and Authenticate Google Cloud SDK
+Open your terminal and run the following commands:
+
+```bash
+# Authenticate with your Google Cloud account
+gcloud auth login
+
+# Set the GCP project
+gcloud config set project {project-name}
+```
+
+## Step 2: Create a Docker Repository on GCP (Optional)
+If you have a docker repository in `GCP` you can skip this step. If not run the following commands to create one:
+
+```bash
+# Create a Docker repository on GCP
+gcloud artifacts repositories create {repository-name} \
+    --repository-format docker \
+    --location us-west1 \
+    --description "Docker repository for storing images of Canopy."
+```
+
+## Step 3: Pull and Tag Canopy Docker Image
+We'll start by fetching the official `Canopy` image from GitHub Packages, and then we'll apply a tag to prepare the image
+for pushing it to Google Cloud Platform (`GCP`). You can access all the available images [here](https://github.com/pinecone-io/canopy/pkgs/container/canopy).
+
+```bash
+# Pull the Canopy Docker image
+docker pull ghcr.io/pinecone-io/canopy:{canopy-version}
```

Comment:
This should probably be `:latest`, and you can mention that we also have individual version images

Reply:
I think latest is not a good practice in general for deployments. (It is handy to check the functionality though.) More context here: https://vsupalov.com/docker-latest-tag/#:~:text=You%20should%20avoid%20using%20the,apart%20from%20the%20image%20ID.